HIRS_Utils/src/main/java/hirs/tpm/eventlog/uefi/UefiGuid.java

```diff
@@ -0,0 +1,415 @@
+import java.util.UUID;
+
+import hirs.utils.HexUtils;
+
+/**
+ * Class to process GUID per the UEFI specification
+ * GUIDs are essentially UUID as defined by RFC-1422, however Microsoft refers to GUIDS.
+ */
+public class UefiGuid {
+    /** number of 100ns intervals since UUID Epoch. */
+    private static final long UUID_EPOCH_INTERVALS =  0x01b21dd213814000L;
+    /** used for conversion to uuid time. */
+    private static final int UUID_EPOCH_DIVISOR = 10000;
+    /** guid byte array. */
+    private byte[] guid = new byte[UefiConstants.SIZE_16 ];
+    /** UUID object. */
+    private UUID uuid;
+
+    /**
+     * UefiGUID constructor.
+     * @param guidBytes byte array holding a valid guid.
+     */
+    public UefiGuid(final byte[] guidBytes) {
+      System.arraycopy(guidBytes, 0, guid, 0, UefiConstants.SIZE_16);
+      uuid = processGuid(guidBytes);
+    }
+
+ /** Converts a GUID with a byte array to a RFC-1422 UUID object.
+  * Assumes a MS format and converts to Big Endian format used by most others , including Linux
+  * Matched uuids found in /sys/firmware/efi/efivars on Centos 7.
+  */
+private static UUID processGuid(final byte[] guid) {
+   byte[] msb1 = new byte[UefiConstants.SIZE_4];
+   System.arraycopy(guid, 0, msb1, 0, UefiConstants.SIZE_4);
+   byte[] msb1r = HexUtils.leReverseByte(msb1);
+   byte[] msb2 = new byte[UefiConstants.SIZE_4];
+   System.arraycopy(guid, UefiConstants.OFFSET_4, msb2, 0, UefiConstants.SIZE_4);
+   byte[] msb2r = HexUtils.leReverseByte(msb2);
+   byte[] msb2rs = new byte[UefiConstants.SIZE_4];
+   System.arraycopy(msb2r, 0, msb2rs, UefiConstants.OFFSET_2, UefiConstants.SIZE_2);
+   System.arraycopy(msb2r, UefiConstants.OFFSET_2, msb2rs, 0, UefiConstants.SIZE_2);
+   byte[] msbt = new byte[UefiConstants.SIZE_8];
+   System.arraycopy(msb1r, 0, msbt, 0, UefiConstants.SIZE_4);
+   System.arraycopy(msb2rs, 0, msbt, UefiConstants.OFFSET_4, UefiConstants.SIZE_4);
+   long msbl = new BigInteger(msbt).longValue();
+   byte[] lsb = new byte[UefiConstants.SIZE_8];
+   System.arraycopy(guid, UefiConstants.OFFSET_8, lsb, 0, UefiConstants.SIZE_8);
+   long lsbl = new BigInteger(lsb).longValue();
+   UUID tmpUuid = new UUID(msbl, lsbl);
+   return tmpUuid;
+ }
+/**
+ * Returns the standard GUID length.
+ * @return guid length
+ */
+public static int getGuidLength() {
+    return UefiConstants.SIZE_16;
+}
+/**
+ * Returns a String that represents a specification name referenced by the EFI_CONFIGURATION_TABLE
+ * VendorGUID field.  For structure of EFI_CONFIGURATION_TABLE type, the UEFI specification
+ * has set of GUIDs published that represent standards that one can find further information on
+ * the configuration table being referenced.
+ * Refer to section 4.6 of UEFI spec v 2.8, page 101.
+ *
+ * @return A String of major UUID parameters
+ */
+@SuppressWarnings("checkstyle:methodlength")
+public String getVendorTableReference() {
+
+ String vendorRef = uuid.toString().toLowerCase();
```

Comment:
By create another string and not just use the statement uuid.toString().toLowerCase() in the switch condition?

Reply:
OK.